geode-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/set/AbstractSMoveIntegrationTest.java

```diff
@@ -80,6 +81,17 @@ public void givenMoreThanFourArguments_returnsWrongNumberOfArgumentsError() {
             .hasMessageContaining("ERR wrong number of arguments for 'smove' command");
   }
 
+  @Test
+  public void testSmove_returnsWrongType_whenWrongTypeIsUsed() {
+    jedis.set("not-a-set", "value");
+    assertThatThrownBy(() -> jedis.smove("not-a-set", "some-set", "foo"))
+        .hasMessage("WRONGTYPE " + RedisConstants.ERROR_WRONG_TYPE);
+
+    jedis.hset("not-another-set", "field", "value");
+    assertThatThrownBy(() -> jedis.smove("not-another-set", "some-set", "foo"))
+        .hasMessage("WRONGTYPE " + RedisConstants.ERROR_WRONG_TYPE);
+  }
```

Comment:
Looks like the Redis tests are failing when a destination key of the wrong type is specified (vs. the source key).  Could we write tests for that as well?

Reply:
Yes, I've already fixed that and added a test :)